src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendStep.java

```diff
@@ -165,17 +203,22 @@ public Step newInstance(Map<String, Object> arguments) {
             }
 
             if (arguments.containsKey("applicationId")) {
-                Object applicationId = arguments.get("applicationId");
+                String applicationId = (String) arguments.get("applicationId");
 
-                if (applicationId != null) {
-                    step.setApplicationId((String) applicationId);
+                if (applicationId != null && !step.isCreateAppIfNotExist()) {
+                    step.setApplicationId(applicationId);
+                } else if (step.isCreateAppIfNotExist()) {
+                    if (applicationId == null || applicationId.isEmpty()) {
+                        step.setApplicationId(applicationId);
+                    } else {
+                        throw new IllegalArgumentException("If createAppIfNotExist is true, Application ID must not be set");
+                    }
                 } else {
                     throw new IllegalArgumentException("Application ID must be set.");
```

Comment:
not part of this PR but, are fields that throw this exception supposed to be required? They do not get validated if they are not part of the arguments